packages/imperative/src/events/src/EventUtils.ts

```diff
@@ -0,0 +1,213 @@
+ * - event type determination
+ * - subscription creation and setting callbacks
+ * - event writing
+ */
+export class EventUtils {
+
+    /**
+     * Retrieves a list of supported applications from configuration.
+     *
+     * @static
+     * @returns {string[]} List of application names.
+     */
+    public static getListOfApps(): string[] {
+        const extendersJson = ConfigUtils.readExtendersJson();
+        return ["Zowe", ...Object.keys(extendersJson.profileTypes)];
+    }
+
+    /**
+     * Won't throw an error if user entered a valid appName
+     *
+     * @static
+     * @param {string} appName - The name of the application.
+     */
+    public static validateAppName(appName: string): void {
+        const appList = this.getListOfApps();
+        if (appList.includes(appName)) return;
+        // Performing `appList.find(app => app.includes(appName))` will allow for "tags" (or suffixes) coming from `getListOfApps()`
+        // However, we do not want this behavior because it will allow partial application names to be used
+        // Hence why we should probably match the application name with the exact profileType in `extenders.json`
+        throw new ImperativeError({
+            msg: `Application name not found: ${appName}. Please use an application name from the list:\n- ${appList.join("\n- ")}`
+        });
+    }
+
+    /**
+     * Determines if the specified event name is associated with a user event.
+     *
+     * @param {string} eventName - The name of the event.
+     * @return {boolean} True if it is a user event, otherwise false.
+     */
+    public static isUserEvent(eventName: string): boolean {
+        return Object.values<string>(ZoweUserEvents).includes(eventName);
+    }
+
+    /**
+     * Determines if the specified event name is associated with a shared event.
+     *
+     * @param {string} eventName - The name of the event.
+     * @return {boolean} True if it is a shared event, otherwise false.
+     */
+    public static isSharedEvent(eventName: string): boolean {
+        return Object.values<string>(ZoweSharedEvents).includes(eventName);
+    }
+
+    /**
+     * Retrieve the event contents form disk
+     *
+     * @internal This is not intended for application developers
+     * @param eventFilePath The path to the event file
+     * @returns The object representing the Event
+     */
+    public static getEventContents(eventFilePath: string): IEventJson {
+        try {
+            return JSON.parse(fs.readFileSync(eventFilePath).toString());
+        } catch (err) {
+            throw new ImperativeError({ msg: `Unable to retrieve event contents: Path: ${eventFilePath}` });
+        }
+    }
+
+    /**
+     * Determines the directory path for storing event files based on the event type and application name.
+     *
+     * @param {string} appName - The name of the application.
+     * @return {string} The directory path.
+     */
+    public static getEventDir(appName: string): string {
```

Comment:
I believe `@throws` can be added to the typedoc for all functions in `EventUtils`, excluding `writeEvent`